### PR TITLE
Issue #4626: prevent php error when no localization plugin available

### LINKED ISSUE #4626: Error: Call to a member function init() on null in view->init_localization() on updating core (latest release)
**Description of the bug**

This already _did_ pop up in [another issue](https://github.com/backdrop/backdrop-issues/issues/4604#issuecomment-690296336) while testing the preview release, but we missed the actual [root cause](https://github.com/backdrop/backdrop/pull/3132).

**Steps To Reproduce**

1. Install an outdated Backdrop version (like 1.16.2)
2. Enable the locale and language modules and add any language (translations aren't needed)
3. Replace core directory with the one from 1.17.0 (or 1.16.3)
4. Visit the start page or any other page containing a view - **before** flushing caches or running the updater

**Actual behavior**

`Error: Call to a member function init() on null in view->init_localization() (line 2009 of .../core/modules/views/includes/view.inc).`

**Expected behavior**

No error, of course.

Many people probably run the updater immediately after replacing files, but others may not and the site shouldn't look broken then - even if flushing caches fixes the problem.

People who are running core updates via UI won't get that error, neither people who run `drush updb` or `b updb` right after replacing files.

**Additional information**

It's too late for the 1.16.3/1.17.0 releases, but we could fix it for people who didn't update yet.

Possibly by catching it in function init_localization() - if $this->localization_plugin is still NULL, return instead of trying to act on it?

### PR DESCRIPTION
**Pull request by indigoxela**
Fixes https://github.com/backdrop/backdrop-issues/issues/4626

### CODE REVIEW

File: core/modules/views/includes/view.inc [hunk 1]
```diff
@@ -2003,6 +2009,11 @@ class view {
       // property set to FALSE, signifying localization should not occur.
       if (empty($this->localization_plugin)) {
         $this->localization_plugin = views_get_plugin('localization', 'none');
+        // If it is still NULL not even plugin "none" is available. Only needed
+        // for edge cases like core update.
```
Comment (klonos) [inline core/modules/views/includes/view.inc]: Some punctuation is needed here:

```suggestion
        // If it is still NULL, not even plugin "none" is available. Only needed
        // for edge cases, like core update.
```

See:
- https://www.grammarly.com/blog/conditional-sentences "Use a comma after the if-clause when the if-clause precedes the main clause."
- https://medium.com/@Ediket/when-do-you-use-a-comma-before-including-or-such-as-9b3e1b4f7af3 for adding a comma before things like "including", or "such as", or "like".

File: core/modules/views/includes/view.inc [hunk 2]
```diff
@@ -1992,6 +1992,12 @@ class view {
 
   /**
    * Find and initialize the localization plugin.
+   *
+   * @return bool
+   *   Whether the active plugin does handle translation or not. FALSE for the
+   *   the plugin "none" or if no plugin was available.
```
Comment (klonos) [inline core/modules/views/includes/view.inc]:
```suggestion
   *   the plugin "none", or if no plugin was available.
```

See: https://www.grammarly.com/blog/comma-before-or

> Always place a comma before or when it begins an independent clause, but if it begins a dependent clause, don’t.

File: core/modules/views/includes/view.inc (hunk 1 above)
Comment (indigoxela) [reply core/modules/views/includes/view.inc]: I'll just add your commas, I don't want to actually learn punctuation rules in a foreign language. No need for a English course here. :rofl:

File: core/modules/views/includes/view.inc [hunk 3]
```diff
@@ -1992,6 +1992,12 @@ class view {
 
   /**
    * Find and initialize the localization plugin.
+   *
+   * @return bool
+   *   Whether the active plugin does handle translation or not. FALSE for the
+   *   the plugin "none", or if no plugin was available.
```
Comment (klonos) [inline core/modules/views/includes/view.inc]: Remove extra "the" (sorry I missed that in my initial review)

```suggestion
   *   plugin "none", or if no plugin was available.
```

File: core/modules/views/includes/view.inc [hunk 4]
```diff
@@ -2003,6 +2009,11 @@ class view {
       // property set to FALSE, signifying localization should not occur.
       if (empty($this->localization_plugin)) {
         $this->localization_plugin = views_get_plugin('localization', 'none');
+        // If it is still NULL, not even plugin "none" is available. Only
+        // needed for edge cases, like core update.
```
Comment (klonos) [inline core/modules/views/includes/view.inc]: "needed" still fits in the previous line, w/o exceeding the 80 char limit.

```suggestion
        // If it is still NULL, not even plugin "none" is available. Only needed
        // for edge cases, like core update.
```

Comment (indigoxela) [reply core/modules/views/includes/view.inc]: Problem: then it's 81 chars. :wink:

File: core/modules/views/includes/view.inc (hunk 3 above)
Comment (indigoxela) [reply core/modules/views/includes/view.inc]: Oops, right, the the...

File: core/modules/views/includes/view.inc (hunk 4 above)
Comment (klonos) [reply core/modules/views/includes/view.inc]: No, it's not:

Screen Shot 2020-09-30 at 5 32 00 pm

Comment (indigoxela) [reply core/modules/views/includes/view.inc]: I'm not sure how to deal with such passion regarding an integer (80). Better not argue...

**Review by klonos — APPROVED** [review]
LGTM 👍

**Comment by ghost** [comment]
Thanks guys, merged into 1.x and 1.17.x.

**Pull request merged**